Stack overflow in ImportSuggestions#distinctRefs

Scala version: 3.0.0
CATS library: 2.6.1
JDK: Azul Zulu 11.0.11
SBT: 1.5.4

Compile under Maven, it throws the following exception. The error message does not pinpoint a specific source file that caused the issue so I can't be more specific than this.

```scala
[ERROR] ## Exception when compiling 54 sources to /home/chris/IdeaProjects/CatsTest3/target/classes
java.lang.StackOverflowError
scala.collection.StrictOptimizedLinearSeqOps.loop$3(LinearSeq.scala:266)
scala.collection.StrictOptimizedLinearSeqOps.dropWhile(LinearSeq.scala:268)
scala.collection.StrictOptimizedLinearSeqOps.dropWhile$(LinearSeq.scala:264)
scala.collection.immutable.List.dropWhile(List.scala:79)
dotty.tools.dotc.typer.ImportSuggestions.distinctRefs(ImportSuggestions.scala:275)
dotty.tools.dotc.typer.ImportSuggestions.distinctRefs$(ImportSuggestions.scala:23)
dotty.tools.dotc.typer.Typer.distinctRefs(Typer.scala:103)
dotty.tools.dotc.typer.ImportSuggestions.distinctRefs(ImportSuggestions.scala:275)
dotty.tools.dotc.typer.ImportSuggestions.distinctRefs$(ImportSuggestions.scala:23)
dotty.tools.dotc.typer.Typer.distinctRefs(Typer.scala:103)
... last stack frame above repeated many times ...
...
```

When compile under sbt, it reports a number of errors. The last of which is the following. Then the console hangs at compile incremental with the timer keeps ticking:

```scala
[error] 457 | } yield passwordOk
[error] | ^
[error] |Could not find an instance of FlatMap for cats.Id.
[error] |I found:
[error] |
[error] | cats.Invariant.catsStdCommutativeMonadForTuple11[A0, A1, A2, A3, A4, A5, A6, A7
[error] | ,
[error] | A8, A9]
[error] |
[error] |But method catsStdCommutativeMonadForTuple11 in trait NTupleMonadInstances1 does not match type cats.FlatMap[cats.Id].
[error] -- [E008] Not Found Error: xxx.scala:48:25
[error] 48 | import scala.reflect.runtime.universe
[error] | ^^^^^^^^^^^^^^^^^^^^^
[error] | value runtime is not a member of reflect

| => root / Compile / compileIncremental 457s
```

--- a/compiler/src/dotty/tools/dotc/typer/ImportSuggestions.scala
+++ b/compiler/src/dotty/tools/dotc/typer/ImportSuggestions.scala
@@ -273,11 +273,14 @@ trait ImportSuggestions:
   /** The `ref` parts of this list of pairs, discarding subsequent elements that
    *  have the same String part. Elements are sorted by their String parts.
    */
-  extension (refs: List[(TermRef, String)]) def distinctRefs(using Context): List[TermRef] = refs match
-    case (ref, str) :: refs1 =>
-      ref :: refs1.dropWhile(_._2 == str).distinctRefs
-    case Nil =>
-      Nil
+  extension (refs: List[(TermRef, String)]) def distinctRefs(using Context): List[TermRef] =
+    val buf = new mutable.ListBuffer[TermRef]
+    var last = ""
+    for (ref, str) <- refs do
+      if last != str then
+        buf += ref
+        last = str
+    buf.toList
 
   /** The best `n` references in `refs`, according to `compare`
    *  `compare` is a partial order. If there's a tie, we take elements

